owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/handlers.go

```diff
@@ -17,21 +18,44 @@ func HealthCheck(c echo.Context) error {
 func GetTicket(c echo.Context) error {
 	id := c.Param("id")
 	userDataQuery := map[string]interface{}{"userID": id}
-	userDataResult, err := db.GetUserData(userDataQuery)
+	sessionID, err := c.Cookie("sessionIDa5")
 	if err != nil {
-		// could not find this user in MongoDB (or MongoDB err connection)
-		return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error finding this UserID."})
+		return err
 	}
+	token, err := jwt.Parse(sessionID.Value, func(token *jwt.Token) (interface{}, error) {
+		// Don't forget to validate the alg is what you expect:
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
 
-	format := c.QueryParam("format")
-	if format == "json" {
-		return c.JSON(http.StatusOK, map[string]string{
-			"result":   "success",
-			"username": userDataResult.Username,
-			"ticket":   userDataResult.Ticket,
-		})
+		// hmacSampleSecret is a []byte containing your secret, e.g. []byte("my_secret_key")
+		return []byte("secret"), nil
+	})
+
+	if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
+
+		userDataResult, err := db.GetUserData(userDataQuery)
+		if err != nil {
+			// could not find this user in MongoDB (or MongoDB err connection)
+			return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error finding this UserID."})
+		}
+
+		if claims["name"] != userDataResult.Username {
+			return c.JSON(http.StatusForbidden, map[string]string{"result": "error", "details": "I wont give you any error messages so you dont have hints to access my application"})
```

Comment:
😄 That's a pretty good message! You can check [this document](https://www.owasp.org/index.php/Error_Handling#Generic_error_messages) for generic messages later on!